### PR TITLE
Bumped kotlin and coroutines version

### PR DESCRIPTION
**Pull request by gmariotti**
- Updated Kotlin to 1.2.51
- Updated coroutines to 0.23.4
- Refactored code to be more Kotlin-like

### CODE REVIEW

File: vertx-lang-kotlin-coroutines/src/main/java/io/vertx/kotlin/coroutines/VertxCoroutine.kt [hunk 1]
```diff
@@ -195,7 +192,7 @@ class ReceiveChannelHandler<T> constructor(context : Context) : ReceiveChannel<T
  * @param capacity the channel buffering capacity
  */
 fun <T> ReadStream<T>.toChannel(vertx : Vertx, capacity : Int = 256) : ReceiveChannel<T> {
-  return toChannel(vertx.getOrCreateContext(), capacity)
```
Comment (vietj) [inline vertx-lang-kotlin-coroutines/src/main/java/io/vertx/kotlin/coroutines/VertxCoroutine.kt]: please let the original method `getOrCreateContext()`

File: vertx-lang-kotlin-coroutines/src/main/java/io/vertx/kotlin/coroutines/VertxCoroutine.kt [hunk 2]
```diff
@@ -237,22 +234,18 @@ private class ChannelReadStream<T>(val context: Context,
...
     val ret = super.offerInternal(element)
     // Not great - fix this
-    if (ret.toString().equals("OFFER_SUCCESS")) {
-      size++;
-      if (isFull) {
-        stream.pause();
-      }
+    if (ret.toString() == "OFFER_SUCCESS") {
+      size++
+      if (isFull) stream.pause()
```
Comment (vietj) [inline vertx-lang-kotlin-coroutines/src/main/java/io/vertx/kotlin/coroutines/VertxCoroutine.kt]: don't remove braces

File: vertx-lang-kotlin-coroutines/src/main/java/io/vertx/kotlin/coroutines/VertxCoroutine.kt [hunk 3]
```diff
@@ -237,22 +234,18 @@ private class ChannelReadStream<T>(val context: Context,
...
 
   override fun pollInternal(): Any? {
     val ret = super.pollInternal()
     // Not great - fix this
-    if (!ret.toString().equals("POLL_FAILED") && !(ret is Closed<*>)) {
-      if (--size < capacity / 2) {
-        stream.resume()
-      }
+    if (ret.toString() != "POLL_FAILED" && ret !is Closed<*>) {
+      if (--size < capacity / 2) stream.resume()
```
Comment (vietj) [inline vertx-lang-kotlin-coroutines/src/main/java/io/vertx/kotlin/coroutines/VertxCoroutine.kt]: don't remove braces

File: vertx-lang-kotlin-coroutines/src/main/java/io/vertx/kotlin/coroutines/VertxCoroutine.kt [hunk 4]
```diff
@@ -268,7 +261,7 @@ private class ChannelReadStream<T>(val context: Context,
  * @param capacity the channel buffering capacity
  */
 fun <T> WriteStream<T>.toChannel(vertx : Vertx, capacity : Int = 256) : SendChannel<T> {
-  return toChannel(vertx.getOrCreateContext(), capacity)
+  return toChannel(vertx.orCreateContext, capacity)
```
Comment (vietj) [inline vertx-lang-kotlin-coroutines/src/main/java/io/vertx/kotlin/coroutines/VertxCoroutine.kt]: let the method call

**Comment by gmariotti** [comment]
@vietj rolled back all changes related to your comments.

File: vertx-lang-kotlin-coroutines/src/main/java/io/vertx/kotlin/coroutines/VertxCoroutine.kt [hunk 5]
```diff
@@ -128,7 +128,7 @@ suspend fun <T> Future<T>.await(): T = when {
  */
 class ReceiveChannelHandler<T> constructor(context : Context) : ReceiveChannel<T>, Handler<T> {
 
-  constructor(vertx : Vertx) : this(vertx.getOrCreateContext())
+  constructor(vertx : Vertx) : this(vertx.orCreateContext)
```
Comment (vietj) [inline vertx-lang-kotlin-coroutines/src/main/java/io/vertx/kotlin/coroutines/VertxCoroutine.kt]: use `getOrCreateContext()`

File: vertx-lang-kotlin-coroutines/src/main/java/io/vertx/kotlin/coroutines/VertxCoroutine.kt [hunk 6]
```diff
@@ -327,11 +324,9 @@ private class ChannelWriteStream<T>(val context: Context,
  * It uses the Vert.x context event loop.
  *
  * This is necessary if you want to execute coroutine synchronous operations in your handler
- *
- * @param block the coroutine code
  */
 fun Vertx.dispatcher() : CoroutineDispatcher {
-  return getOrCreateContext().dispatcher()
+  return orCreateContext.dispatcher()
```
Comment (vietj) [inline vertx-lang-kotlin-coroutines/src/main/java/io/vertx/kotlin/coroutines/VertxCoroutine.kt]: use `getOrCreateContext()`

**Comment by vietj** [comment]
another couple of usage of `getOrCreateContext()` and we are good, can you also shoot an upgrade for `vertx-codetrans` that uses Kotlin ?

**Comment by vietj** [comment]
I think you just need to make a library version bump and check tests pass

**Comment by gmariotti** [comment]
if you're fine with this PR, I would merge it. For the problems I'm encountering with Kotlin 1.2.61, I opened #77

**Pull request merged**